### DISCUSSION TITLE
Please take care with your doco

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#17242] [Q&A General] [answered]
Most of you are better coders and designers than i'll ever be, but please take lots of care and time with your documentation.  This is especially important in framework objects, because "modules" classes tend to be documented from a user's perspective, so are generally easy to understand (but not always).  Please take care to use well-written and explanatory english, not things like `params.addParam<Real>("coef", 1, "The coefficient");`.  Please take care to provide well-written markdown documentation, not just barely-filled stubs like
```
!syntax description blah
!syntax parameters blah
!syntax inputs blah
!syntax children blah
```
Please include examples so users don't have to trawl the test files.

**Comment by GiudGiud** [comment, marked as answer]
Fully agree with this and have been improving the framework doco continuously along with others. 
Going to mark this as closed even though this is an always ongoing effort, just to make it easier to find un-answered topics.
